File: front-end/src/app/service-models/app/v1/app-users/login/login-user-s-m.ts
import { CoinManagementServiceModelBase } from '../../../base/coin-management-service-model-base';
import { RoleTypeSM } from '../../../enums/role-type-s-m.enum';
import { LoginStatusSM } from '../../../enums/login-status-s-m.enum';

export class LoginUserSM extends CoinManagementServiceModelBase<number> {
    roleType!: RoleTypeSM;
    loginId!: string;
    firstName!: string;
    middleName!: string;
    lastName!: string;
    emailId!: string;
    passwordHash!: string;
    phoneNumber!: string;
    profilePicturePath!: string;
    isPhoneNumberConfirmed!: boolean;
    isEmailConfirmed!: boolean;
    loginStatus!: LoginStatusSM;
    dateOfBirth!: Date;
}

File: backend/payroll-backend/Components/Siffrum.Web.Payroll.ServiceModels/v1/AppUsers/Login/LoginUserSM.cs
using Siffrum.Web.Payroll.ServiceModels.Enums;

namespace Siffrum.Web.Payroll.ServiceModels.v1.AppUsers.Login
{
    public class LoginUserSM : Base.SiffrumPayrollServiceModelBase<int>
    {
        public LoginUserSM()
        {
        }
        public RoleTypeSM RoleType { get; set; }
        public string LoginId { get; set; }
        public string FirstName { get; set; }
        public string MiddleName { get; set; }
        public string LastName { get; set; }
        public string EmailId { get; set; }

        [IgnorePropertyOnWrite(AutoMapConversionType.Dm2SmOnly)]
        public string PasswordHash { get; set; }
        public string PhoneNumber { get; set; }
        public string ProfilePicturePath { get; set; }

        //[ConvertFilePathToUri(SourcePropertyName = nameof(ProfilePicturePath))]
        //public string ProfilePicturePathUrl { get; set; }
        public bool IsPhoneNumberConfirmed { get; set; }
        public bool IsEmailConfirmed { get; set; }
        public LoginStatusSM LoginStatus { get; set; }
        public DateTime DateOfBirth { get; set; }
    }
}

